ui/packages/consul-ui/app/services/repository/auth-method.js

```diff
@@ -0,0 +1,26 @@
+import RepositoryService from 'consul-ui/services/repository';
+import statusFactory from 'consul-ui/utils/acls-status';
+import isValidServerErrorFactory from 'consul-ui/utils/http/acl/is-valid-server-error';
+import { PRIMARY_KEY, SLUG_KEY } from 'consul-ui/models/auth-method';
+
+const isValidServerError = isValidServerErrorFactory();
+const status = statusFactory(isValidServerError, Promise);
+const MODEL_NAME = 'auth-method';
+
+export default class AuthMethodService extends RepositoryService {
+  getModelName() {
+    return MODEL_NAME;
+  }
+
+  getPrimaryKey() {
+    return PRIMARY_KEY;
+  }
+
+  getSlugKey() {
+    return SLUG_KEY;
+  }
+
+  status(obj) {
+    return status(obj);
+  }
```

Comment:
Another little note to self here, we can probably stop injecting `Promise` here since we stopped using `rsvp` a while ago in favour of native Promises.

Reply:
If it's going to be removed soon then best to leave it in these acls repos.